src/parser/core/modules/DoTs.ts

```diff
@@ -79,74 +64,63 @@ export abstract class DoTs extends Analyser {
-			start: event.timestamp - this.statusDurations[statusID],
-			end: event.timestamp,
-			types: ['untargetable'],
-		})
-
-		// Wait for when the status would typically drop without clipping:
-		//  clipping a dot early isn't as problematic if it would just push it into invuln time.
-		this.addTimestampHook(
-			Math.min(
-				event.timestamp + this.statusDurations[statusID] + clip,
-				this.parser.pull.timestamp + this.parser.pull.duration,
-			),
-			({timestamp}) => {
-				clip -= this.invulnerability.getDuration({
-					start: event.timestamp,
-					end: timestamp,
-					types: ['invulnerable'],
-				})
-
-				// Clamp clips at 0 - less than that is downtime, which is handled by the checklist requirement.
-				this.clips[statusID] = (this.clips[statusID] ?? 0) + Math.max(0, clip)
-			},
-		)
-
-		lastApplication[statusID] = event.timestamp
+		const clip = status.duration - (event.timestamp - trackedStatusOnTarget.lastApplied)
+		// Cap clip at 0 - less than that is downtime, which is handled by the checklist requirement
+		trackedStatusOnTarget.totalClipping += Math.max(0, clip)
+		trackedStatusOnTarget.lastApplied = event.timestamp
 	}
 
 	private onComplete() {
 		this.addChecklistRules()
-		this.addClippingSuggestions(this.clips)
+		this.addClippingSuggestions()
 	}
 
 	// These two functions are helpers for submodules and should be used but not overridden
-	public getUptimePercent(statusID: number) {
-		const status = this.data.getStatus(statusID)
+	protected getUptimePercent(statusId: number) {
+		const status = this.data.getStatus(statusId)
 		if (status == null) { return 0 }
 
 		const statusUptime = this.statuses.getUptime(status, this.actors.foes)
-		const fightDuration = this.parser.currentDuration - this.invulnerability.getDuration({types: ['invulnerable']})
+		const fightDuration = this.parser.pull.duration - this.invulnerability.getDuration({types: ['invulnerable']})
 		return (statusUptime / fightDuration) * 100
 	}
 
-	public getClippingAmount(statusID: number) {
+	protected getClippingAmount(statusId: number) {
 		// This normalises clipping as seconds clipped per minute,
 		// since some level of clipping is expected and we need tiers that work for both long and short fights
-		const fightDurationMillis = (this.parser.currentDuration - this.invulnerability.getDuration({types: ['invulnerable']}))
+		const fightDurationMillis = (this.parser.pull.duration - this.invulnerability.getDuration({types: ['invulnerable']}))
 		if (fightDurationMillis <= 0) { return 0 }
 
-		const clipSecsPerMin = Math.round(((this.clips[statusID] ?? 0) * SECONDS_PER_MINUTE) / fightDurationMillis)
+		const statusApplications = this.statusApplications.get(statusId)
+		if (statusApplications == null) { return 0 }
+
+		const totalClipping = Array.from(statusApplications.values()).reduce((clip, target) => clip + target.totalClipping, 0)
+		const clipSecsPerMin = Math.round(totalClipping / (fightDurationMillis / MILLISECONDS_PER_MINUTE))
```

Comment:
I know, I know, it's a breaking change - but while you're in here (or perhaps as seperate PR) it'd be good to move this module to milliseconds, too. It's long term, but I'm trying to get _all_ time values in ms for the sake of consistency.

Reply:
So good news and bad news on this.  It already was returning milliseconds per minute, so all the consumers were basically always returning major.  Fixed.